src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -324,9 +333,40 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
          * allocated in a context should also allows us to identify
          * virtual/stream-ordered CUDA allocations. */
         mem_info->type = UCS_MEMORY_TYPE_CUDA_MANAGED;
+
+        UCS_STATIC_ASSERT(sizeof(pref_loc) == 4);
+
+        cu_err =
+            cuMemRangeGetAttribute((void*)&pref_loc, sizeof(pref_loc),
+                                   CU_MEM_RANGE_ATTRIBUTE_PREFERRED_LOCATION,
+                                   (CUdeviceptr)address, length);
+        if ((cu_err == CUDA_SUCCESS) && (pref_loc != CU_DEVICE_INVALID)) {
+            if (pref_loc == CU_DEVICE_CPU) {
+                mem_info->sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
+            } else {
+                uct_cuda_base_get_sys_dev(pref_loc, &mem_info->sys_dev);
+                if (mem_info->sys_dev == UCS_SYS_DEVICE_ID_UNKNOWN) {
+                    ucs_diag("preferred location %d (%p...%p) unrecognized",
```

Comment:
do we expect this to happen in normal flow? If no, I'd make it `ucs_warn`, otherwise `ucs_debug` or even `ucs_trace`